src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,450 @@
+                var frameworkTransitivePackages = frameworkPackages.transitivePacakges;
+
+                //Filter the packages for outdated
+                if (outdated)
+                {
+                    frameworkTopLevelPackages = frameworkTopLevelPackages.Where(p => !p.autoReference && p.resolvedVersion < p.suggestedVersion);
+                    frameworkTransitivePackages = frameworkTransitivePackages.Where(p => p.resolvedVersion < p.suggestedVersion);
+                }
+
+                //If no packages exist for this framework, print the
+                //appropriate message
+                if (!frameworkTopLevelPackages.Any() && !frameworkTransitivePackages.Any())
+                {
+                    frameworkMessage.Append(": ");
+                    Console.WriteLine(string.Format(frameworkMessage.ToString() + Strings.ListPkg_NoPackagesForFramework, frameworkPackages.framework));
+                }
+                else
+                {
+                    //Print name of the framework
+                    Console.WriteLine(string.Format(frameworkMessage.ToString(), frameworkPackages.framework));
+
+                    //Print top-level packages
+                    if (frameworkTopLevelPackages.Any())
+                    {
+                        var autoRefWithinPackagesList = false;
+                        PackagesTable(frameworkTopLevelPackages, false, outdated, out autoRefWithinPackagesList);
+                        autoReferenceFound = autoReferenceFound || autoRefWithinPackagesList;
+                    }
+
+                    //Print transitive pacakges
+                    if (transitive && frameworkTransitivePackages.Any())
+                    {
+                        var autoRefWithinPackagesList = false;
+                        PackagesTable(frameworkTransitivePackages, true, outdated, out autoRefWithinPackagesList);
+                        autoReferenceFound = autoReferenceFound || autoRefWithinPackagesList;
+                    }
+                }
+            }            
+        }
+
+        /// <summary>
+        /// Given a list of packages, this function will print them in a table
+        /// </summary>
+        /// <param name="packages">The list of packages</param>
+        /// <param name="printingTransitive">Whether the function is printing transitive
+        /// packages table or not</param>
+        /// <param name="outdated"></param>
+        /// <param name="autoRefWithinPackagesList"></param>
+        /// <returns>The table as a string</returns>
+        private void PackagesTable(IEnumerable<PRPackage> packages, bool printingTransitive, bool outdated, out bool autoRefWithinPackagesList)
+        {
+            var autoReferenceFound = false;
+
+            if (!packages.Any())
+            {
+                autoRefWithinPackagesList = false;
+                return;
+            }
+
+            List<Tuple<string, ConsoleColor>> tableToPrint;
```

Comment:
list of tuples ? any reason you can't use a dictionary or lookup?

Reply:
I need to preserve order